plugins/autosuspend.koplugin/main.lua

```diff
@@ -141,34 +153,55 @@ function AutoSuspendWidget:addToMainMenu(menu_items)
                 ok_text = _("Set timeout"),
                 title_text = _("Timeout in minutes"),
                 callback = function(autosuspend_spin)
-                    G_reader_settings:saveSetting("auto_suspend_timeout_seconds", autosuspend_spin.value * 60)
-                    -- NOTE: Will only take effect after a restart, as we don't have a method to set this live...
+                    local autosuspend_timeout_seconds = autosuspend_spin.value * 60
+                    self.auto_suspend_sec = autosuspend_timeout_seconds
+                    G_reader_settings:saveSetting("auto_suspend_timeout_seconds", autosuspend_timeout_seconds)
                     UIManager:show(InfoMessage:new{
-                        text = _("This will take effect on next restart."),
+                        text = T(_("The system will automatically suspend after %1 minutes of inactivity."),
+                            string.format("%.2f", autosuspend_timeout_seconds/60)),
                     })
+                    self:_deprecateLastTask()
+                    self:_start()
+                end
+            }
+            UIManager:show(autosuspend_spin)
+        end,
+    }
+    menu_items.autoshutdown = {
+        text = _("Autoshutdown timeout"),
+        callback = function()
+            local InfoMessage = require("ui/widget/infomessage")
+            local Screen = Device.screen
+            local SpinWidget = require("ui/widget/spinwidget")
+            local curr_items = G_reader_settings:readSetting("autoshutdown_timeout_seconds") or default_autoshutdown_timeout_seconds
+            local autosuspend_spin = SpinWidget:new {
+                width = Screen:getWidth() * 0.6,
+                value = curr_items / 24 / 60 / 60,
+                -- About a minute, good for testing and battery life fanatics.
+                -- Just high enough to avoid an instant shutdown death scenario.
+                value_min = 0.0007,
+                -- More than three weeks seems a bit excessive if you want to enable authoshutdown,
+                -- even if the battery can last up to three months.
+                value_max = 21,
+                value_hold_step = 3,
+                precision = "%.2f",
+                ok_text = _("Set timeout"),
+                title_text = _("Timeout in days"),
```

Comment:
Why not timeout in hours? Easier to compute that 3 days = 72 hours - than compute that 4 hours is 0.17 days.

Reply:
Perhaps, but except for testing any value under 24 hours is basically useless. And I don't find 2 weeks = 336 hours particularly easy tbh. :-)

Although on second thought it might have some use as a workaround for issues like #3855.